front/lib/client/assistant_builder/deprecated_single_action.ts

```diff
@@ -0,0 +1,18 @@
+import { useMemo } from "react";
+
+import type {
+  AssistantBuilderActionConfiguration,
+  AssistantBuilderState,
+} from "@app/components/assistant_builder/types";
+
+export function getDeprecatedDefaultSingleAction(
+  builderState: AssistantBuilderState
+): AssistantBuilderActionConfiguration | undefined {
+  return builderState.actions[0];
+}
+
+export function useDeprecatedDefaultSingleAction(
+  builderState: AssistantBuilderState
+): AssistantBuilderActionConfiguration | undefined {
+  return useMemo(() => builderState.actions[0], [builderState.actions]);
```

Comment:
made this because otherwise we'd trigger all the effects that depend on the action on every render.